code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingExtension.java

```diff
@@ -897,17 +929,14 @@ private Map<String, Object> getXDMSharedState(final String extensionName, final
     }
 
     private boolean eventIsValid(final Event event) {
-        return event != null && event.getEventData() != null;
+        return (event != null && event.getEventData() != null) || isRequestResetEvent(event);
+    }
+
+    private boolean isRequestResetEvent(final Event event) {
+        return event != null && event.getSource().equals(EventSource.REQUEST_RESET);
     }
 
     private void createMessagingSharedState(final String pushToken, final Event event) {
-        if (StringUtils.isNullOrEmpty(pushToken)) {
-            Log.debug(
-                    MessagingConstants.LOG_TAG,
-                    SELF_TAG,
-                    "Push token is null or empty. Not adding the push token to the shared state.");
-            return;
-        }
         final Map<String, Object> newMessagingState = new HashMap<>();
         newMessagingState.put(MessagingConstants.SharedState.Messaging.PUSH_IDENTIFIER, pushToken);
```

Comment:
Check if `pushToken` is not null or empty before adding to `newMessagingState`. When the push token is cleared, then the Messaging shared state will simply be empty.